File: library/src/main/java/com/dimeno/permission/callback/PermissionCallback.java
package com.dimeno.permission.callback;

/**
 * PermissionCallback
 * Created by wangzhen on 2020/4/15.
 */
public interface PermissionCallback {
    /**
     * all permissions are granted
     *
     * @param permissions permissions
     */
    void onGrant(String[] permissions);

    /**
     * permissions are denied or refused
     *
     * @param deniedPermissions   denied permissions
     * @param neverAskPermissions refused permissions
     */
    void onDeny(String[] deniedPermissions, String[] neverAskPermissions);

    /**
     * permissions not declared in manifest
     *
     * @param permissions permissions
     */
    void onNotDeclared(String[] permissions);
}

File: library/src/main/java/com/wangzhen/permission/callback/PermissionCallback.kt
package com.wangzhen.permission.callback

/**
 * PermissionCallback
 * Created by wangzhen on 2020/4/15.
 */
interface PermissionCallback {
    /**
     * all permissions are granted
     *
     * @param permissions permissions
     */
    fun onGrant(permissions: Array<String>)

    /**
     * permissions are denied or refused
     *
     * @param deniedPermissions   denied permissions
     * @param neverAskPermissions refused permissions
     */
    fun onDeny(deniedPermissions: Array<String>, neverAskPermissions: Array<String>)

    /**
     * permissions not declared in manifest
     *
     * @param permissions permissions
     */
    fun onNotDeclared(permissions: Array<String>)
}